src/java.base/share/classes/sun/net/ext/ExtendedSocketOptions.java

```diff
@@ -167,27 +167,34 @@ protected ExtendedSocketOptions(Set<SocketOption<?>> options) {
 
     private static volatile ExtendedSocketOptions instance;
 
-    public static final ExtendedSocketOptions getInstance() { return instance; }
+    public static ExtendedSocketOptions getInstance() {
+        if (instance != null) {
```

Comment:
May be it's worth to avoid reading `volatile` field twice?

Reply:
Hello @turbanoff, do you mean why read it twice - once here and once inside the `synchronized` block?

Reply:
Yes. Once here and once inside `synchronized` block.
Reading `volatile` fields cost _something_ on some architectures, so I think we could optimize it a bit.

Reply:
Hello @turbanoff, the double read is necessary to correctly avoid any race conditions and is a typical strategy used in cases like these. I am not aware of any other alternate more performant strategy, for code like this.

Reply:
Let me be more clear: I think that it's enough to have only 2 `volatile` field reads _in worst case_. We can use local variable to avoid more than needed reads.
Current code can perform read 4 times _in worst case_: twice outside `synchronized` block and twice inside `synchronized` block.

There are many examples of similar code in the JDK:
https://github.com/openjdk/jdk/blob/master/src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Util.java#L48
https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/lang/StackFrameInfo.java#L125

Reply:
Ah! of course, I now see what you mean. I misunderstood your previous comment. I'll update this PR shortly.

Reply:
I've now updated this PR which this suggested change.